quil-rs/src/instruction/calibration.rs

```diff
@@ -20,46 +20,42 @@ pub trait CalibrationSignature {
 
 #[derive(Clone, Debug, Default, PartialEq)]
 pub struct Calibration {
+    pub identifier: CalibrationIdentifier,
     pub instructions: Vec<Instruction>,
-    pub modifiers: Vec<GateModifier>,
-    pub name: String,
-    pub parameters: Vec<Expression>,
-    pub qubits: Vec<Qubit>,
 }
 
 impl Calibration {
     /// Builds a new calibration definition.
-    ///
-    /// # Errors
-    ///
-    /// Returns an error if the given name isn't a valid Quil identifier.
     pub fn new(
-        name: &str,
-        parameters: Vec<Expression>,
-        qubits: Vec<Qubit>,
+        identifier: CalibrationIdentifier,
         instructions: Vec<Instruction>,
-        modifiers: Vec<GateModifier>,
     ) -> Result<Self, IdentifierValidationError> {
-        validate_identifier(name)?;
         Ok(Self {
+            identifier,
             instructions,
-            modifiers,
-            name: name.to_string(),
-            parameters,
-            qubits,
         })
     }
 }
 
+impl CalibrationSignature for Calibration {
```

Comment:
From @MarquessV : this should not be needed anymore if the `CalibrationIdentifier` can serve as the signature

Reply:
That is true, I don't know of a good way to make the generics work with just a `CalibrationIdentifier` struct. I agree that it makes sense to formalize the concept of a calibration identifier w/ a struct and defer to it to implement the signature logic. They serve different purposes, so it's not entirely redundant, and the core logic doesn't need to be duplicated between either, so I'm good with this.